components/abstractions/src/main/java/com/microsoft/kiota/RequestInformation.java

```diff
@@ -268,26 +271,31 @@ public void setResponseHandler(@Nonnull ResponseHandler responseHandler) {
     /**
      * Sets the request body to be a binary stream.
      * @param value the binary stream
+     * @throws IOException when the stream cannot be read.
      * @deprecated use {@link #setStreamContent(InputStream, String)} instead.
      */
     @Deprecated
-    public void setStreamContent(@Nonnull final InputStream value) {
+    public void setStreamContent(@Nonnull final InputStream value) throws IOException {
         setStreamContent(value, BINARY_CONTENT_TYPE);
     }
 
     /**
      * Sets the request body to be a binary stream.
      * @param value the binary stream
      * @param contentType the content type of the stream.
+     * @throws IOException when the stream cannot be read.
      */
     public void setStreamContent(
-            @Nonnull final InputStream value, @Nonnull final String contentType) {
+            @Nonnull final InputStream value, @Nonnull final String contentType)
```

Comment:
I think this method should either accept a content length or have an overload that accepts a content length.

```java
public void setStreamContent(@Nonnull final InputStream value, @Nonnull final String contentType, final Long contentLength) {...}
```

Reply:
instead of making changes to request information that are specific to java, and have undesired effects (source breaking or wrapping exceptions), could we handle that in the request adapter? (see my other comment)

Reply:
Adding an overload isn't a source breaking change unless I'm mistaken. If it works in the request adapter as well, I'm fine with it. But I'd argue that content information belongs in the request information since it's part of the request.

Reply:
right, but it'd be the only language with a content length field. I'm not sure why we'd do that only for Java?

Reply:
While I do agree that adding the check at the RequestAdapter makes sense as the least interruptive method for handling this, I also agree with @calebkiage when we consider that we already include contentLength in a lot of the methods for java specifically due to the languages need for this value across many operations when dealing with input streams in particular. I'm not familiar with the other languages and their handling of this, but with java it seems the content-length appears frequently and thus is a valid candidate for a value that should be stored in the requestInformation specifically for the Java language.